src/form/Checkbox/Checkbox.tsx

```diff
@@ -121,6 +123,16 @@ export const Checkbox: FC<CheckboxProps & CheckboxRef> = forwardRef(
     const pathLength = useMotionValue(0);
     const opacity = useTransform(pathLength, [0.05, 0.15], [0, 1]);
 
+    // If the checkbox is inside a dialog, the animation will not work.
+    // This is a workaround to force the animation to work by triggering
+    // a re-render once after initial mount
+    const [_, setForceAnimation] = useState<boolean>(false);
+    useEffect(() => {
+      if (checked || intermediate) {
+        setForceAnimation(true);
+      }
+    }, []); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
we could probably use something like `useEffectOnce` but we're not currently including `react-use` package so I'm just disabling the eslint rule here instead